dll/cpl/sysdm/environment.c

```diff
@@ -1827,6 +1853,10 @@ EnvironmentDlgProc(HWND hwndDlg,
 {
     PENVIRONMENT_DIALOG_DATA DlgData;
     DlgData = (PENVIRONMENT_DIALOG_DATA)GetWindowLongPtr(hwndDlg, DWLP_USER);
+    WCHAR szBuffer[MAX_STR_LENGTH] = {0};
```

Comment:
```suggestion
    WCHAR szBuffer[MAX_STR_LENGTH + UNLEN + 1];
```
Resized to contain both ``szFormat`` and ``szUserName``